lib/src/main/java/com/diffplug/spotless/RoundedStep.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2024 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+import java.io.Serializable;
+
+public interface RoundedStep extends Serializable {}
```

Comment:
The intent here is to be a marker interface?

Reply:
Yeah.

Reply:
@Goooler My YAGNI senses are tingling! I'm wondering if you can clarify why this interface was introduced, and if we can get away with inlining `Serializable` instead?

Otherwise, this all LGTM. :+1:

Reply:
I was thinking that we can do some checks based on a common step type instead of names here

https://github.com/diffplug/spotless/blob/224f8f96df3ad42cac81064a0461e6d4ee91dcaf/testlib/src/main/java/com/diffplug/spotless/StepHarnessBase.java#L52-L55

Reply:
That's a good idea, but it's hard to get to that state. I figured it was probably meant as a testing thing, we can remove it when we finish #1274

Reply:
Sounds good to me either way! 👍